specs/core/1_beacon-chain-misc.md

```diff
@@ -0,0 +1,252 @@
+        SHARD_SLOTS_PER_EPOCH * EPOCHS_PER_SHARD_PERIOD
+    )
+    first_slot_in_last_crosslink = state.current_crosslinks[receipt_proof.shard].start_epoch * SHARD_SLOTS_PER_EPOCH
+    gindex = concat_generalized_indices(
+        get_generalized_index_of_crosslink_header(0),
+        GeneralizedIndex(get_generalized_index(ShardBlockHeader, 'state_root')),
+        compute_historical_state_generalized_index(receipt_slot, first_slot_in_last_crosslink),
+        GeneralizedIndex(get_generalized_index(ShardState, 'receipt_root'))
+    )
+    assert verify_merkle_proof(
+        leaf=hash_tree_root(receipt_proof.receipt),
+        proof=receipt_proof.proof,
+        index=gindex,
+        root=state.current_crosslinks[receipt_proof.shard].data_root
+    )
+    for delta in receipt_proof.receipt:
+        if get_current_epoch(state) < state.validators[delta.index].withdrawable_epoch:
+            increase_amount = (
+                state.validators[delta.index].effective_balance * delta.reward_coefficient // REWARD_COEFFICIENT_BASE
+            )
+            increase_balance(state, delta.index, increase_amount)
+            decrease_balance(state, delta.index, delta.block_fee)
+    state.next_shard_receipt_period[receipt_proof.shard] += 1
+    proposer_index = get_beacon_proposer_index(state)
+    increase_balance(state, proposer_index, Gwei(get_base_reward(state, proposer_index) // MINOR_REWARD_QUOTIENT))
+```
+
+## Changes
+
+### Phase 0 container updates
+
+Add the following fields to the end of the specified container objects.
+
+#### `BeaconState`
+
+```python
+class BeaconState(Container):
+    # Period committees
+    period_committee_roots: Vector[Hash, PERIOD_COMMITTEE_ROOT_LENGTH]
+    next_shard_receipt_period: Vector[uint64, SHARD_COUNT]
+```
+
+`period_committee_roots` values are initialized to `Bytes32()` (empty bytes value).
+`next_shard_receipt_period` values are initialized to `compute_epoch_at_slot(PHASE_1_FORK_SLOT) // EPOCHS_PER_SHARD_PERIOD`.
+
+#### `BeaconBlockBody`
+
+```python
+class BeaconBlockBody(Container):
+    shard_receipt_proofs: List[ShardReceiptProof, MAX_SHARD_RECEIPT_PROOFS]
+```
+
+`shard_receipt_proofs` is initialized to `[]`.
+
+### Persistent committees
+
+Run `update_period_committee` immediately before `process_final_updates`:
+
+```python
+# begin insert @update_period_committee
+    update_period_committee(state)
+# end insert @update_period_committee
+def update_period_committee(state: BeaconState) -> None:
+    """
+    Updates period committee roots at boundary blocks.
+    """
+    if (get_current_epoch(state) + 1) % EPOCHS_PER_SHARD_PERIOD != 0:
+        return
+
+    period = (get_current_epoch(state) + 1) // EPOCHS_PER_SHARD_PERIOD
+    committees = Vector[CompactCommittee, SHARD_COUNT]([
+        committee_to_compact_committee(
+            state,
+            get_period_committee(state, Shard(shard), Epoch(get_current_epoch(state) + 1)),
+        )
+        for shard in range(SHARD_COUNT)
```

Comment:
`SHARD_COUNT` removed?

Reply:
All of the phase 1 stuff is in a state of disarray. Thus the turning off of phase 1 tests in CI 😬 

This is noted in the comment for the PR and again in the release notes so will leave as is for now